### ISSUE TITLE
Worksheet ready, but not opening

### ISSUE THREAD

**Issue by dsche-cyber** [millejoh/emacs-ipython-notebook#749]
## Problem description
Issue same as #480 
Open a .ipynb in doom emacs. `C-c C-o`. Choose parent dir of opened .ipynb.

There are three different cases that appear:
1.) `Notebook .ipynb pending open" Retry? (y or n)` after y nothing happens (ksysguard shows a new process jupyter-notebook)
2.) Notebook does not open in any buffer, instead minibuffer prints Worksheet is ready.
3.) For some notebooks it works (the notebook opens in a new buffer)

## Steps to reproduce the problem
After ein:notebook-open, click on Open for some notebooks.

## Info asked for in #480
After `M-x ein:log-pop-to-all-buffer`

In `M-x ein:log-pop-to-request-buffer` I get:
```
[debug] request--callback: executing success
[debug] request--callback: executing complete
[debug] /tmp/curl-trace unreadable
```

The debug report:
## System info:

```cl
("EIN system info"
 :emacs-version
 "GNU Emacs 27.1 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.22, cairo version 1.17.3)
 of 2020-08-28"
 :emacs-bzr-version nil
 :window-system x
 :emacs-variant nil
 :os
 (:uname
  "Linux dsche-pc 5.8.11-1-MANJARO #1 SMP PREEMPT Wed Sep 23 14:35:40 UTC 2020 x86_64 GNU/Linux
"
  :lsb-release
  "LSB Version:	n/a
Distributor ID:	ManjaroLinux
Description:	Manjaro Linux
Release:	20.1.1
Codename:	Mikah
")
 :notebook
 "Name: notebook
Version: 6.1.4
Summary: A web-based notebook environment for interactive computing
Home-page: http://jupyter.org
Author: Jupyter Development Team
Author-email: jupyter@googlegroups.com
License: BSD
Location: /usr/lib/python3.8/site-packages
Requires: jinja2, tornado, pyzmq, argon2-cffi, ipython-genutils, traitlets, jupyter-core, jupyter-client, nbformat, nbconvert, ipykernel, Send2Trash, terminado, prometheus-client
Required-by: widgetsnbextension, jupyterlab, jupyterlab-server
"
 :ipython "7.18.1
"
 :image-types
 (svg png gif tiff jpeg xpm xbm pbm)
 :image-types-available
 (svg png gif tiff jpeg xpm xbm pbm)
 :request
 (:backend curl)
 :ein
 (:version nil
           :source-dir "/home/dsche/.emacs.d/.local/straight/build/ein/")
 :lib
 ((:name "websocket"
         :path "~/.emacs.d/.local/straight/build/websocket/websocket.elc"
         :featurep t
         :version-var websocket-version
         :version "1.12")
  (:name "anaphora"
         :path "~/.emacs.d/.local/straight/build/anaphora/anaphora.elc"
         :featurep t
         :version-var nil
         :version nil)
  (:name "request"
         :path "~/.emacs.d/.local/straight/build/request/request.elc"
         :featurep t
         :version-var request-version
         :version "0.3.0")
  (:name "deferred"
         :path "~/.emacs.d/.local/straight/build/deferred/deferred.elc"
         :featurep t
         :version-var deferred:version
         :version "0.5.0")
  (:name "polymode"
         :path "~/.emacs.d/.local/straight/build/polymode/polymode.elc"
         :featurep t
         :version-var nil
         :version nil)
  (:name "exec-path-from-shell"
         :path
         "~/.emacs.d/.local/straight/build/exec-path-from-shell/exec-path-from-shell.elc"
         :featurep t
         :version-var nil
         :version nil)
  (:name "dash"
         :path "~/.emacs.d/.local/straight/build/dash/dash.elc"
         :featurep t
         :version-var nil
         :version nil)
  (:name "with-editor"
         :path "~/.emacs.d/.local/straight/build/with-editor/with-editor.elc"
         :featurep t
         :version-var nil
         :version nil)))
```

**Comment by dickmao** [comment]
Uploading the problematic .ipynb here, if that's possible, would help me reproduce the problem.

**Comment by dsche-cyber** [comment]
In this folder here I tried to open `calibrating-qubits-pulse.ipynb`:
[](https://github.com/qiskit-community/qiskit-textbook/tree/master/content/ch-quantum-hardware)

**Comment by dickmao** [comment]
Indeed I observed the same failure-to-launch with doom + emacs-27.1.  Below is my attempt with doom+ emacs-26.3, which at least opens the notebook.  I don't plan on investigating this in the next two weeks.  Sorry, and thanks for your interest.

![Peek 2020-10-14 12-10](https://user-images.githubusercontent.com/7578770/96016381-ab3e7a80-0e16-11eb-915e-bbbfa2e3b5c4.gif)

**Comment by dickmao** [comment]
Thanks for reporting the bug.  fe63100b should help.

**Comment by dsche-cyber** [comment]
The first time I tried to open it, it worked, but now it does not.

In  `M-x ein:log-pop-to-request-buffer `I get:
```
[debug] request--callback: executing success
[debug] request--callback: executing complete
[debug] /tmp/curl-trace unreadable
[debug] request--curl-callback: event finished

[debug] request--callback: UNPARSED

[debug] request--callback: PARSED

[debug] request--callback: executing success
[debug] request--callback: executing complete
[debug] /tmp/curl-trace unreadable
```

**Comment by dickmao** [comment]
`grep -- gc-operation /home/dsche/.emacs.d/.local/straight/build/ein/*` should return nothing.

Here I open the notebook, run `ein:stop`, then reopen the notebook.
![doom](https://user-images.githubusercontent.com/7578770/96576412-27fea800-12a0-11eb-83e5-e8ea48272231.gif)

**Comment by dsche-cyber** [comment]
Indeed you command returns nothing.
For you the notebook opens automatically, for me not.
Finally I got it: I needed to manually open it (C-x b, and choose the buffer, I think this is not intended behaviour).

Moreover I get a Warning  after C-c C-o.
Also Latex text is not displaying.

![ein_not_working](https://user-images.githubusercontent.com/66935312/96702245-ddf5ef00-1391-11eb-9d33-a1610ab9dc8c.gif)

**Comment by dickmao** [comment]
If it weren't already obvious, I think your EIN experience under your current doom configuration may not be worth it.

EIN cannot handle any undo-mode other than the default.  It [deftly sidesteps undo-tree-mode](https://github.com/millejoh/emacs-ipython-notebook/blob/fbc1d99a11ddceb5fd90a42956bbf0b3681326e5/lisp/poly-ein.el#L317-L319) but I did not foresee undo-fu-mode, and noticed it's not easily sidestepped without messing with a user's global preferences.  EIN can only warn "Disabling undo for undo-fu-only-undo" and consequently undo won't work at all.

There is also an outstanding memory leak for single-line ipynb files in #740 .

As for the central issue of "Notebook pending open!  Retry?", I suspect an error is getting swallowed.  You'd need to `M-: (setq debug-on-error t)` to get emacs to stop and backtrace when it happens.

Latex rendering is a known deficiency https://github.com/millejoh/emacs-ipython-notebook/projects/3
